syntax = "proto3";

message protoSpeedSSL{
    double vx = 1;
    double vy = 2;
    double vw  = 3;
    bool front  = 4;
    bool chip  = 5;
    bool charge  = 6;
    double kickStrength = 7 ;
    bool dribbler = 8;
    double dribSpeed = 9;
};

message protoOdometrySSL{
    double x = 1;
    double y = 2;
    double w = 3;
    bool hasBall = 4;
    double kickLoad = 5;
    double battery = 6;
    int32 count = 7;
    double vision_x = 8;
    double vision_y = 9;
    double vision_w = 10;
    double vx = 11;
    double vy = 12;
    double vw = 13;
};

message protoPositionSSL{
    double x = 1;
    double y = 2;
    double w = 3;
    enum PosType{
        unknown = 0;
        source = 1;
        driveToTarget = 2;
        rotateInPoint = 3;
        rotateOnSelf = 4;
        stop = 5;
    }
    PosType posType = 4;
    bool front  = 5;
    bool chip  = 6;
    bool charge  = 7;
    double kickStrength = 8 ;
    bool dribbler = 9;
    double dribSpeed = 10;
    bool resetOdometry = 11;
    double min_speed = 12;
    double max_speed = 13;
};